nixos/modules/services/torrent/transmission.nix

```diff
@@ -5,48 +5,49 @@ with lib;
 let
   cfg = config.services.transmission;
   apparmor = config.security.apparmor.enable;
-
-  homeDir = cfg.home;
-  downloadDirPermissions = cfg.downloadDirPermissions;
-  downloadDir = "${homeDir}/Downloads";
-  incompleteDir = "${homeDir}/.incomplete";
-
-  settingsDir = "${homeDir}/config";
-  settingsFile = pkgs.writeText "settings.json" (builtins.toJSON fullSettings);
-
-  # for users in group "transmission" to have access to torrents
-  fullSettings = { umask = 2; download-dir = downloadDir; incomplete-dir = incompleteDir; } // cfg.settings;
-
-  preStart = pkgs.writeScript "transmission-pre-start" ''
-    #!${pkgs.runtimeShell}
-    set -ex
-    cp -f ${settingsFile} ${settingsDir}/settings.json
-  '';
+  # TODO: switch to configGen.json once RFC0042 is implemented
+  settingsFile = pkgs.writeText "settings.json" (builtins.toJSON cfg.settings);
+  settingsDir = ".config/transmission-daemon";
+  makeAbsolute = base: path:
+    if builtins.match "^/.*" path == null
+    then base+"/"+path else path;
 in
 {
+  imports = [
+    (mkRemovedOptionModule [ "services" "transmission" "port" ]
+      "Instead, use the option `services.transmission.settings.rpc-port'.")
+  ];
+
   options = {
     services.transmission = {
-      enable = mkOption {
-        type = types.bool;
-        default = false;
-        description = ''
-          Whether or not to enable the headless Transmission BitTorrent daemon.
+      enable = mkEnableOption ''
+        Whether or not to enable the headless Transmission BitTorrent daemon.
 
-          Transmission daemon can be controlled via the RPC interface using
-          transmission-remote or the WebUI (http://localhost:9091/ by default).
+        Transmission daemon can be controlled via the RPC interface using
+        transmission-remote, the WebUI (http://${cfg.settings.rpc-bind-address}:${toString cfg.settings.rpc-port}/ by default),
+        or other clients like stig or tremc.
 
-          Torrents are downloaded to ${downloadDir} by default and are
-          accessible to users in the "transmission" group.
-        '';
-      };
+        Torrents are downloaded to ${cfg.settings.download-dir} by default and are
+        accessible to users in the "transmission" group.
+      '';
 
-      settings = mkOption {
+      settings = mkOption rec {
+        # TODO: switch to types.config.json as prescribed by RFC0042 once it's implemented
         type = types.attrs;
+        apply = attrs:
+          let super = recursiveUpdate default attrs; in
+          super // {
+            download-dir   = makeAbsolute cfg.home super.download-dir;
+            incomplete-dir = makeAbsolute cfg.home super.incomplete-dir;
+          };
         default =
           {
-            download-dir = downloadDir;
-            incomplete-dir = incompleteDir;
+            download-dir = "${cfg.home}/Downloads";
```

Comment:
I personally would have preferred it to be `downloads` - lower case, but since that was the default from before I guess it'll be better to keep it that way? Not sure what would be best here...

Reply:
The XDG desktop specification also defaults to `Downloads` for `XDG_DOWNLOAD_DIR`.